GrabOffsetInc/Dec can be done with force push/pull buttons.

lumayaa opened the pull request:
This means u no longer need these binds: GrabOffsetIncrease / GrabOffsetDecrease, as it works on your push/pull buttons.

lumayaa committed:
GrabOffsetInc/Dec can be done with force push/pull buttons.

slash-fr commented:
I like the idea, and it (mostly) works.

A few notes though:

## 1. Grabbing several entities

One can actually grab **several** entities at the same time.

In that case, the new push/pull behavior also works, but the push/pull buttons revert to their usual behavior as soon as **one** of the entities is dropped.

I would suggest turning `gentity_t.client->Lmd.grabbing` into a counter (instead of a toggle).
- `int grabbing` instead of `qboolean grabbing`
- `grabbing++` instead of `grabbing = qtrue`
- `grabbing--` instead of `grabbing = qfalse`
- and `ent->client->Lmd.grabbing > 0` in `ClientThink_real()`


## 2. CancelGrab
Currently, `/CancelGrab` doesn't "revert" the push/pull buttons to their standard behavior.  
We should also update the `grabbing` counter (or flag) in `Cmd_CancelGrab_f()`.  

## 3. Minor hardening
When releasing a grabbed entity, I recommend updating `grabbing` *before* the following `if` block:  
```c
if (!Lmdp_EditEntity(ent))
{
    Disp(player, "^1Entity failed to respawn.");
    return 0;
}
```
to make sure we're not "stuck" in "GrabOffsetIncrease/Decrease" mode if the entity *does* fail to respawn.

lumayaa committed:
Grabbing overhaul.

Made grabbing is a counter instead of a boolean.
Added grabbing to Cancelgrab.
Moved grabbing before "Entity failed to respawn.".
